lib/classification_lifecycle.rb

```diff
@@ -14,6 +14,12 @@ def queue(action)
       raise ClassificationNotPersisted.new(message)
     end
     ClassificationWorker.perform_async(classification.id, action)
+    if classification.complete?
+      classification.subject_ids.each do |sid|
+        ClassificationCountWorker.perform_async(sid, classification.workflow_id)
+        RetirementWorker.perform_async(sid, classification.workflow_id)
```

Comment:
We can't guarantee ordering of worker execution here so if a classification that is 1 off retirement happens then the retirement worker may execute before the counter bump and we will end up with an extra classification. What about just chaining the `RetirementWorker` from the `ClassificationCountWorker` after bumping the count?